akka-stream-tests/src/test/scala/akka/stream/FusingSpec.scala

```diff
@@ -79,12 +90,23 @@ class FusingSpec extends StreamSpec {
       verify(fused, modules = 1, downstreams = 6)
     }
 
+    "properly fuse an async FusedGraph that has been extended (no AsyncBoundary)" in {
+      val src = Future(Fusing.aggressive(graph(false)))
+      val fused = Fusing.aggressive(Source.fromFutureGraph(src).to(Sink.head))
+      asyncVerify(fused, modules = 1, downstreams = 1)
+    }
+
     "properly fuse a FusedGraph that has been extended (with AsyncBoundary)" in {
       val src = Fusing.aggressive(graph(true))
       val fused = Fusing.aggressive(Source.fromGraph(src).to(Sink.head))
       verify(fused, modules = 2, downstreams = 6)
     }
 
+    "properly fuse an async FusedGraph that has been extended (with AsyncBoundary)" in {
+      val src = Future(Fusing.aggressive(graph(true)))
+      val fused = Fusing.aggressive(Source.fromFutureGraph(src).to(Sink.head))
+      asyncVerify(fused, modules = 1, downstreams = 1)
+    }
```

Comment:
added test coverage is nice, but maybe put those as well as the java mapAsyncMerge above in a separate PR that details what they are rather than have them in this PR?